File: adrop-ads-example-ios-objective-c/BannerViewController.m
//
//  BannerViewController.m
//  adrop-ads-example-ios-objective-c
//
//  Created by Leo on 1/17/24.
//

#import "BannerViewController.h"

@implementation BannerViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    // Create and load standard banner ad.
    AdropBanner *bannerView = [[AdropBanner alloc] initWithUnitId:@"PUBLIC_TEST_UNIT_ID_375_80" contextId: @""];
    bannerView.frame = self.adContainerNormal.bounds;
    bannerView.delegate = self;
    [bannerView load];
    [self.adContainerNormal addSubview: bannerView];

    // Create and load carousel banner ad.
    AdropBanner *carouselBannerView = [[AdropBanner alloc] initWithUnitId:@"PUBLIC_TEST_UNIT_ID_CAROUSEL" contextId: @""];
    carouselBannerView.frame = self.adContainerNormal.bounds;
    carouselBannerView.delegate = self;
    [carouselBannerView load];
    [self.adContainerCarousel addSubview: carouselBannerView];
}

#pragma mark - AdropBannerDelegate

// Called when ad is successfully loaded.
- (void)onAdReceived:(AdropBanner *)banner {
    NSLog(@"onAdReceived");
}

// Called when ad is clicked.
- (void)onAdClicked:(AdropBanner *)banner {
    NSLog(@"onAdClickced");
}

// Called when ad failed to load.
- (void)onAdFailedToReceive:(AdropBanner *)banner error:(AdropErrorCode)error {
    NSLog(@"onAdFailedToReceive %ld", (long)error);
}
@end

File: adrop-ads-example-ios/BannerViewController.swift
//
//  BannerViewController.swift
//  adrop-ads-example-ios
//
//  Created by Leo on 1/17/24.
//

import UIKit
import AdropAds

class BannerViewController: UIViewController {
    @IBOutlet weak var adContainerNormal: UIView!
    @IBOutlet weak var adContainerCarousel: UIView!

    override func viewDidLoad() {
        super.viewDidLoad()

        // Create and load standard banner ad.
        let banner = AdropBanner(unitId: "PUBLIC_TEST_UNIT_ID_375_80")
        banner.frame = adContainerNormal.bounds
        banner.delegate = self
        banner.load()
        adContainerNormal.addSubview(banner)

        // Create and load carousel banner ad.
        let carouselBanner = AdropBanner(unitId: "PUBLIC_TEST_UNIT_ID_CAROUSEL")
        carouselBanner.frame = adContainerCarousel.bounds
        carouselBanner.delegate = self
        carouselBanner.load()
        adContainerCarousel.addSubview(carouselBanner)
    }
}

// MARK: - AdropBannerDelegate

extension BannerViewController: AdropBannerDelegate {
    // Called when ad is successfully loaded.
    func onAdReceived(_ banner: AdropAds.AdropBanner) {
        print("onAdReceived")
    }

    // Called when ad is clicked.
    func onAdClicked(_ banner: AdropAds.AdropBanner) {
        print("onAdClickced")
    }

    // Called when ad failed to load.
    func onAdFailedToReceive(_ banner: AdropAds.AdropBanner, _ error: AdropAds.AdropErrorCode) {
        print("onAdFailedToReceive \(AdropErrorCodeToString(code: error))")
    }
}
